go/vt/vttablet/tabletmanager/rpc_query.go

```diff
@@ -24,11 +24,50 @@ import (
 	"vitess.io/vitess/go/sqltypes"
 	"vitess.io/vitess/go/vt/log"
 	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vterrors"
 
 	querypb "vitess.io/vitess/go/vt/proto/query"
 	tabletmanagerdatapb "vitess.io/vitess/go/vt/proto/tabletmanagerdata"
+	"vitess.io/vitess/go/vt/proto/vtrpc"
 )
 
+// analyzeExecuteFetchAsDbaMultiQuery reutrns 'true' when at least one of the queries
+// in the given SQL has a `/*vt+ allowZeroInDate=true */` directive.
+func analyzeExecuteFetchAsDbaMultiQuery(sql string, parser *sqlparser.Parser) (queries []string, parseable bool, countCreate int, allowZeroInDate bool, err error) {
+	queries, err = parser.SplitStatementToPieces(sql)
+	if err != nil {
+		return nil, false, 0, false, err
+	}
+	if len(queries) == 0 {
+		return nil, false, 0, false, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "no statements found in query: %s", sql)
+	}
+	parseable = true
+	for _, query := range queries {
+		// Some of the queries we receive here are legitimately non-parseable by our
+		// current parser, such as `CHANGE REPLICATION SOURCE TO...`. We must allow
+		// them and so we skip parsing errors.
+		stmt, err := parser.Parse(query)
+		if err != nil {
+			parseable = false
+			continue
+		}
+		switch stmt.(type) {
+		case *sqlparser.CreateTable, *sqlparser.CreateView:
+			countCreate++
+		default:
+		}
```

Comment:
why do we have only `Create`? Should this not be a `DDL` type query check?

Reply:
It's debatable. Right now the intention of `--batch-size` is to speed up the creation of new schemas, which may contain hundreds or thousands of tables/views. It _could_ technically extend to any DDL, but then that creates a situation where we'd want to track the result of any change, per change. Also, if there's `ALTER`s, then what's the benefit of speeding up a few seconds worth of batching when the migration could take hours to run.

Reply:
The idea was also to limit it to the smallest thing possible here to avoid having to say that multi statements are supported with existing fetch. 

One we have a new api we’ll also move this and this will error then on any multi statement.